lib/backend/etcdbk/etcd.go

```diff
@@ -687,19 +692,40 @@ func (b *EtcdBackend) syncLegacyPrefix(ctx context.Context) error {
 	b.Infof("Migrating Teleport etcd data from legacy prefix %q to configured prefix %q", legacyDefaultPrefix, b.cfg.Key)
 	defer b.Infof("Teleport etcd data migration complete")
 
+	var errs []error
+
 	// Now we know that legacy prefix has some data newer than the configured
 	// prefix. Migrate it over to configured prefix.
 	//
-	// Start with deleting existing prefix data.
+	// First, let's backup the data under configured prefix, in case the
+	// migration kicked in by mistake.
+	backupPrefix := b.backupPrefix(b.cfg.Key)
+	b.Debugf("Backup everything under %q to %q", b.cfg.Key, backupPrefix)
+	for _, kv := range prefixData.Kvs {
+		// Replace the prefix.
+		key := backupPrefix + strings.TrimPrefix(string(kv.Key), b.cfg.Key)
+		b.Debugf("Copying %q -> %q", kv.Key, key)
+		if _, err := b.client.Put(ctx, key, string(kv.Value)); err != nil {
+			errs = append(errs, trace.WrapWithMessage(err, "failed copying %q to %q: %v", kv.Key, key, err))
```

Comment:
Shouldn't we stop migration if this occurs?

Reply:
should we crash the teleport process then, or let it proceed with stale/missing data in custom prefix?

Reply:
crashing makes sense, it means we can't write to etcd.
updated.